flycheck.el

```diff
@@ -8060,12 +8060,14 @@ for more information about the custom directories."
   :safe #'flycheck-string-list-p
   :package-version '(flycheck . "29"))
 
-(defun flycheck-eslint-config-exists-p ()
-  "Whether there is an eslint config for the current buffer."
-  (let* ((executable (flycheck-find-checker-executable 'javascript-eslint))
-         (exitcode (and executable (call-process executable nil nil nil
-                                                 "--print-config" "."))))
-    (eq exitcode 0)))
+(defun flycheck-eslint-config-error ()
+  "String of eslint config error for the current buffer."
+  (let ((executable (flycheck-find-checker-executable 'javascript-eslint))
+        (stderr-file (flycheck-temp-file-system "eslint-config-errors")))
+    (call-process executable nil `(nil ,stderr-file) nil "--print-config" ".")
+    (with-temp-buffer
+      (insert-file-contents stderr-file)
+      (buffer-string))))
```

Comment:
I'm not a big fan of returning an empty string to indicate success, and I'd rather not depend too strongly on eslint writing to stderr. Could we keep the return code check, and return a cons of `status . error-message`?

Reply:
Yes this is indeed suboptimal. I fill amend this PR.

Reply:
Few possible pointers:
- If config is missing `eslint --print-config .` will not print in `stderr` but `stdout` and exit with exit code 1.
- I'm not sure how long we can relay on this behavior from `eslint`, it looks that every mayor version break everything. Maybe we just need to check if output from command is JSON and use better but more generic message "Config file: missing or incorrect".
- This implemented right will close GH-1175
- This may slow even more opening of any JS file, so we may add to the documentation how to disable this check for users who don't want it, as described in GH-1129.